satpy/resample.py

```diff
@@ -479,61 +490,96 @@ def precompute(self, mask=None, radius_of_influence=None, epsilon=0,
             self.resampler.get_neighbour_info(mask=mask)
             self.save_neighbour_info(cache_dir, mask=mask, **kwargs)
 
-    def _apply_cached_indexes(self, cached_indexes, persist=False):
-        """Reassign various resampler index attributes."""
-        # cacheable_dict = {}
-        for elt in ['valid_input_index', 'valid_output_index',
-                    'index_array', 'distance_array']:
-            val = cached_indexes[elt]
-            if isinstance(val, tuple):
-                val = cached_indexes[elt][0]
-            elif isinstance(val, np.ndarray):
-                val = da.from_array(val, chunks=CHUNK_SIZE)
-            elif persist and isinstance(val, da.Array):
-                cached_indexes[elt] = val = val.persist()
-            setattr(self.resampler, elt, val)
+    def _apply_cached_index(self, val, idx_name, persist=False):
+        """Reassign resampler index attributes."""
+        if isinstance(val, np.ndarray):
+            val = da.from_array(val, chunks=CHUNK_SIZE)
+        elif persist and isinstance(val, da.Array):
+            val = val.persist()
+        setattr(self.resampler, idx_name, val)
+        return val
+
+    def _check_numpy_cache(self, cache_dir, mask=None,
+                           **kwargs):
+        """Check if there's Numpy cache file and convert it to zarr."""
+        fname_np = self._create_cache_filename(cache_dir,
+                                               prefix='resample_lut-',
+                                               mask=mask, fmt='.npz',
+                                               **kwargs)
+        fname_zarr = self._create_cache_filename(cache_dir, prefix='nn_lut-',
+                                                 mask=mask, fmt='.zarr',
+                                                 **kwargs)
+
+        if os.path.exists(fname_np) and not os.path.exists(fname_zarr):
+            import warnings
+            warnings.warn("Using Numpy files as resampling cache is "
+                          "deprecated.")
+            LOG.warning("Converting resampling LUT from .npz to .zarr")
+            zarr_out = xr.Dataset()
+            with np.load(fname_np, 'r') as fid:
+                for idx_name, coord in NN_COORDINATES.items():
+                    zarr_out[idx_name] = (coord, fid[idx_name])
+
+            # Write indices to Zarr file
+            zarr_out.to_zarr(fname_zarr)
```

Comment:
Should the npz cache files be deleted after the conversion ?

Reply:
I was thinking about that, but was hesitant to do that (automatically) if the user happens to have another streams still using them.

Reply:
ok, fine by me

Reply:
What do you mean by "stream"? If there is an active process using the file then deleting it shouldn't be a problem (it should still be held in memory). I suppose we are assuming that two different versions of satpy may be using the same resampling cache? I guess it would be a problem then.

I'm fine with leaving them there, but I don't think it is necessary.

Reply:
"Stream" = "processing chain". Someone might have another process running an older Satpy for whatever reason, e.g. for comparison reasons before upgrading an operational chain to the latest versions.